### PR TITLE
add map for Set (Fixes #342)

### LINKED ISSUE #342: add 'map' for Set

### PR DESCRIPTION
**Pull request by sledorze**

### CODE REVIEW

File: src/Set.ts [hunk 1]
```diff
@@ -31,6 +31,21 @@ export const some = <A>(x: Set<A>, predicate: Predicate<A>): boolean => {
   return found
 }
 
+/**
+  * Projects a Set through a function
+  *  @function */
+export const map = <A, B>(x: Set<A>, bset: Setoid<B>, f: (x: A) => B): Set<B> => {
```
Comment (gcanti) [inline src/Set.ts]: Following the convention used so far, shouldn't be

```ts
export const map = <B>(S: Setoid<B>) => <A>(x: Set<A>, f: (x: A) => B): Set<B> => {
  ...
}
```

?

File: src/Set.ts [hunk 2]
```diff
@@ -31,6 +31,21 @@ export const some = <A>(x: Set<A>, predicate: Predicate<A>): boolean => {
...
+  const ismember = member(bset)
+  const r = new Set<B>()
+  x.forEach(e => {
+    const v = f(e)
+    if (!ismember(r)(v)) {
```
Comment (gcanti) [inline src/Set.ts]: Maybe we can do less work here defining `ismember` as

```ts
const r = new Set<B>()
const ismember = member(bset)(r)
```

Comment (sledorze) [reply src/Set.ts]: Indeed, also I'm not sure what to favor between using `forEach`or using an Iterator, do you have any clue?

File: src/Set.ts (hunk 1 above)
Comment (sledorze) [reply src/Set.ts]: ok done

**Pull request merged**